settings.c

```diff
@@ -268,9 +268,50 @@ static void UaServer_FSBE_RemoveEntry(FileSettings *pFS, int index)
         free(pEntry->pszValue);
         pEntry->pszValue = 0;
     }
-    pEntry->type = Invalid;
-    pEntry->parent = -1;
-    pEntry->line = 0;
+
+    // Move all following entries to this entry...
+    for (int i = index; i < pFS->numEntries-1; i++)
+    {
+        Entry *pEntryFrom = &pFS->pEntries[i+1];
+        Entry *pEntryTo = &pFS->pEntries[i];
+
+        *pEntryTo = *pEntryFrom;  // assign all values directly
+
+        pEntryFrom->pszKey = 0;
+        pEntryFrom->pszValue = 0;
+        pEntryFrom->type = Invalid;
+        pEntryFrom->parent = -1;
+        pEntryFrom->line = 0;
+    }
+
+    pFS->numEntries--; // we have now one entry less...
+
+    // be sure the last entry is cleared
+    Entry *pEntryToClear = &pFS->pEntries[pFS->numEntries];
+
+    pEntryToClear->pszKey = 0;
```

Comment:
Should pFS->pEntries be realloc'd?  I assume most implementations do just return the existing one, but it would give the runtime a choice to free up memory at least.  Or maybe that is not intended, seeing that the parent is marked with -1 in this case...

Reply:
Hi Marc, I decided not to realloc because the settings system already keeps track of the "allocated entries" and the "used entries". We just reduce the "usedEntries" now (before, nothing was removed). I decided not to do a realloc because of:
- The normal behavior in health_check is to remove the section and the add the same section again... this would lead into two reallocs and end up with the same size... so this is really an overhead...
- It is very unusual that a large number of entries gets freed; even if some servers will unregister, it is likely the case that these server will re-register later again
So from my point of view, it has no positive effect on doing a realloc.